src/mrinufft/operators/interfaces/gpunufft.py

```diff
@@ -0,0 +1,275 @@
+        )
+
+    def op(self, image, interpolate_data=False):
+        """Compute the masked non-cartesian Fourier transform.
+
+        Parameters
+        ----------
+        image: np.ndarray
+            input array with the same shape as shape.
+        interpolate_data: bool, default False
+            if set to True, the image is just apodized and interpolated to
+            kspace locations. This is used for density estimation.
+
+        Returns
+        -------
+        np.ndarray
+            Non Uniform Fourier transform of the input image.
+        """
+        # Base gpuNUFFT Operator is written in CUDA and C++, we need to
+        # reorganize data to follow a different memory hierarchy
+        # TODO we need to update codes to use np.reshape for all this directly
+        if self.n_coils > 1 and not self.uses_sense:
+            coeff = self.operator.op(
+                np.asarray(
+                    [np.reshape(image_ch.T, image_ch.size) for image_ch in image]
+                ).T,
+                interpolate_data,
+            )
+        else:
+            coeff = self.operator.op(np.reshape(image.T, image.size), interpolate_data)
+            # Data is always returned as num_channels X coeff_array,
+            # so for single channel, we extract single array
+            if not self.uses_sense:
+                coeff = coeff[0]
+        return coeff
+
+    def adj_op(self, coeff, grid_data=False):
+        """Compute adjoint  of non-uniform Fourier Transform.
+
+        Parameters
+        ----------
+        coeff: np.ndarray
+            masked non-uniform Fourier transform 1D data.
+        grid_data: bool, default False
+            if True, the kspace data is gridded and returned,
+            this is used for density compensation
+
+        Returns
+        -------
+        np.ndarray
+            adjoint operator of Non Uniform Fourier transform of the
+            input coefficients.
+        """
+        image = self.operator.adj_op(coeff, grid_data)
+        if self.n_coils > 1 and not self.uses_sense:
+            image = np.asarray([image_ch.T for image_ch in image])
+        else:
+            image = np.squeeze(image).T
+        # The recieved data from gpuNUFFT is num_channels x Nx x Ny x Nz,
+        # hence we use squeeze
+        return np.squeeze(image)
+
+
+class MRIGpuNUFFT(FourierOperatorBase):
+    """Interface for the gpuNUFFT backend.
+
+    Parameters
+    ----------
+    samples: np.ndarray (Mxd)
+        the samples locations in the Fourier domain where M is the number
+        of samples and d is the dimensionnality of the output data
+        (2D for an image, 3D for a volume).
+    shape: tuple of int
+        shape of the image (not necessarly a square matrix).
+    implementation: str 'cpu' | 'gpuNUFFT',
+    default 'cpu'
+        which implementation of NFFT to use.
+    n_coils: int default 1
+        Number of coils used to acquire the signal in case of multiarray
```

Comment:
Inconsistent use of points and capital letters in the documentation: All parameter descriptions should follow the same convention of start with/without capital letter and (not?) finish with points